pulsar-client-cpp/lib/ConsumerImpl.cc

```diff
@@ -919,4 +919,65 @@ void ConsumerImpl::seekAsync(const MessageId& msgId, ResultCallback callback) {
     callback(ResultNotConnected);
 }
 
+void ConsumerImpl::hasMessageAvailableAsync(HasMessageAvailableCallback callback) {
+    if (this->lastMessageIdAvailable()) {
+        callback(ResultOk, true);
+        return;
+    }
+
+    BrokerGetLastMessageIdCallback callback1 = [this, callback](Result result, MessageId messageId) {
+        if (this->lastMessageIdAvailable()) {
+            callback(ResultOk, true);
+        } else {
+            callback(ResultOk, false);
+        }
+    };
+
+    getLastMessageIdAsync(callback1);
+}
+
+void ConsumerImpl::brokerGetLastMessageIdListener(Result res, MessageId messageId,
+                                                  BrokerGetLastMessageIdCallback callback) {
+    if (!callback.empty()) {
```

Comment:
what does empty() do? Can the callback become empty between here and where callback is used?